### PR TITLE
Bug 2209258: Fixes sidebar content length dynamically based on screen size

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2023-07-28 13-04-44
Screenshot from 2023-07-28 13-04-50

### CODE REVIEW

File: packages/shared/src/hooks/useAutoExpand.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const useAutoExpand = () => {
+  const ref = React.useRef(null);
+  const [height, setHeight] = React.useState(0);
+
+  const updateHeight = () => {
+    if (ref.current) {
+      const windowHeight = window.innerHeight;
+      const divTopOffset = ref.current.getBoundingClientRect().top;
+      const availableHeight = windowHeight - divTopOffset;
+      setHeight(availableHeight);
```
Comment (SanjalKatiyar) [inline packages/shared/src/hooks/useAutoExpand.ts:12]: just a nit (can be skipped as well):
```suggestion
      availableHeight !== height && setHeight(availableHeight);
```

File: packages/shared/src/hooks/useAutoExpand.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  };
+
+  React.useEffect(() => {
+    updateHeight(); // Set initial height on mount
```
Comment (SanjalKatiyar) [inline packages/shared/src/hooks/useAutoExpand.ts:17]: won't `ref.current` be `null` during initial mount and then later ref will get attached to the sideBar component ?? (`updateHeight` only runs when ref is non-empty)

Comment (bipuladh) [reply packages/shared/src/hooks/useAutoExpand.ts:17]: Hence we check for ref.current in line 8.

File: packages/shared/src/hooks/useAutoExpand.ts (hunk 1 above)
Comment (bipuladh) [reply packages/shared/src/hooks/useAutoExpand.ts:12]: This should only be triggered when the window size is changed so the guard might not be necessary.

File: packages/shared/src/hooks/useAutoExpand.ts (hunk 2 above)
Comment (SanjalKatiyar) [reply packages/shared/src/hooks/useAutoExpand.ts:17]: I think it will get attached first... due to lifecycle of useEffect...

Comment (SanjalKatiyar) [reply packages/shared/src/hooks/useAutoExpand.ts:17]: nvm ignore this comment...

**Pull request merged**